src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ActivatorUtilitiesTests.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.Extensions.DependencyInjection.Tests
+{
+    public class ActivatorUtilitiesTests
+    {
+        [Fact]
+        public void ShouldUseLongestAvailableConstructorOnlyIfConstructorsHaveTheSamePriority()
+        {
+            var services = new ServiceCollection();
+            services.AddScoped<B>();
+            services.AddScoped<S>();
+            using var provider = services.BuildServiceProvider();
+            var a = new A();
+            var c = new C();
+
+            var instance = ActivatorUtilities.CreateInstance<Creatable>(provider, a, c);
+
+            Assert.Null(instance.B);
+        }
+
+        [Theory]
+        [InlineData(typeof(DefaultConstructorFirst))]
+        [InlineData(typeof(DefaultConstructorLast))]
+        public void ChoosesDefaultConstructorNoMatterOrder(Type instanceType)
+        {
+            var services = new ServiceCollection();
+            using var provider = services.BuildServiceProvider();
+
+            var instance = ActivatorUtilities.CreateInstance(provider, instanceType);
+
+            Assert.NotNull(instance);
+        }
+
+        [Fact]
+        public void ShouldFixIssue_46132()
+        {
+            var services = new ServiceCollection();
+            services.AddScoped<S>();
+            using var provider = services.BuildServiceProvider();
+            var a = new A();
+            var c = new C();
+
+            var instance = ActivatorUtilities.CreateInstance<Creatable>(
+                provider, c, a);
+
+            Assert.NotNull(instance);
+            Assert.Same(a, instance.A);
+            Assert.Same(c, instance.C);
+            Assert.NotNull(instance.S);
+            Assert.Null(instance.B);
+        }
+
+        [Theory]
+        [MemberData(nameof(ActivatorUtilitiesData))]
```

Comment:
```suggestion
        [InlineData(typeof(FakeValidationResult))]
        [InlineData(typeof(FakeValidationResultOps))]
```

It's simpler to just use inline data.